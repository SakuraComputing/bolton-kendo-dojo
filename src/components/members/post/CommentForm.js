import React, { Component } from 'react'

class commentForm extends Component {
    render() {
        return (
            <div>
                
            </div>
        )
    }
}
export default commentForm;
