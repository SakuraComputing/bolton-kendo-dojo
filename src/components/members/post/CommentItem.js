import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deleteComment } from '../../../actions/postActions';

export class CommentItem extends Component {

    onDeleteClick(postId, commentId) {
        this.props.deleteComment(postId, commentId);
    }

    render() {
        const { comment, postId, auth } = this.props;

        return(
            <div className="post-container">
                <div className="comment-outer">
                    <div className="comment-inner-1">
                        <a href="profile.html">
                            <img className="rounded-circle d-none d-md-block"
                                    src={comment.avatar} alt=""/>
                        </a>
                        <br/>
                        <p className="avatar-name">{comment.name}</p>
                        <div className="comment-container">
                        </div>
                    </div>
                    <div className="comment-inner-2">
                        <p className="comment-text">{comment.text}</p>
                    </div>
                    <div className="comment-inner-3">
                        {comment.user === auth.user.id ? (
                            <button onClick={this.onDeleteClick.bind(this, postId, comment._id)}
                                    type="button"
                                    className="button-small button-delete item-delete"
                            >
                            <i className="fas fa-times"></i>
                            Delete Comment
                            </button>
                        ) : null}
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    auth: state.auth
});
 
export default connect(mapStateToProps, {deleteComment})(CommentItem);
