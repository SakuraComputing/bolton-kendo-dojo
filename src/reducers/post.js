import {
    GET_POSTS,
    ADD_POST,
    GET_ERRORS,
    DELETE_POST,
    GET_POST

} from '../actions/types';

const postReducerDefaultState = [];

export default (state = postReducerDefaultState, action) => {
    switch(action.type) {
        
    }
}
