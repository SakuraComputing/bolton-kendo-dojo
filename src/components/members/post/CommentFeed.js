import React, { Component } from 'react';
import CommentItem from './CommentItem';

class commentFeed extends Component {
    render() {
        const {comments, postId} = this.props;

        return comments.map(comment => (
            <CommentItem key={comment._id} comment={comment} postId={postId}/>
        ))
    }
}
export default commentFeed;
