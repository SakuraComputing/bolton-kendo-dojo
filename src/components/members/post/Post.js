import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getPost } from '../../../actions/postActions';
import Spinner from '../../../common/Spinner';
import PostItem from '../posts/PostItem';
import CommentForm from './CommentForm';
import CommentFeed from './CommentFeed';

export class Post extends Component {

    componentDidMount() {
        this.props.getPost(this.props.match.params.id);
    }

    render() {
        const { post, loading } = this.props.post;

        let postContent;

        if(post === null || loading || Object.keys(post).length === 0) {
            postContent = <Spinner />
        } else {
            postContent = (
                <div>
                    <PostItem post={post} showActions={false}/>
                    <CommentForm postId={post._id}/>
                    <CommentFeed postId={post._id} comments={post.comments} />
                </div>
            )
        }
        
        return (
            <div>
                <div className="back-to-feed">
                    <Link to="/feed" className="button-small button-warning">
                        Back To Feed
                    </Link>
                </div>
                {postContent}
            </div>
        )
    }
}
const mapStateToProps = state => ({
    post: state.post
})
export default connect(mapStateToProps, { getPost })(Post);
