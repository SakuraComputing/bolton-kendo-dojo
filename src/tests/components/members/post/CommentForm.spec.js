import React from 'react';
import { shallow } from 'enzyme';
import { CommentForm } from '../../../../components/members/post/CommentForm';

describe('Comment Form Component', () => {
  let commentForm;

  beforeEach(() => {
      commentForm = shallow(<CommentForm />)
  });
  it('should render correctly', () => {
      expect(commentForm).toMatchSnapshot();
  });
})
