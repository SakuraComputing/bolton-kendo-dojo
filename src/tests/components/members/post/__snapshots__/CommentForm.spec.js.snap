// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Form Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card card-info"
      >
        <div
          className="card-header bg-info text-white"
        >
          Make a comment...
        </div>
        <div
          className="card-body"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <TextAreaFieldGroup
                name="text"
                onChange={[Function]}
                placeholder="Reply to post"
                value=""
              />
            </div>
            <button
              className="button-small button-warning"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>,
      "className": "post-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-header bg-info text-white"
          >
            Make a comment...
          </div>,
          <div
            className="card-body"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <TextAreaFieldGroup
                  name="text"
                  onChange={[Function]}
                  placeholder="Reply to post"
                  value=""
                />
              </div>
              <button
                className="button-small button-warning"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>,
        ],
        "className": "card card-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Make a comment...",
            "className": "card-header bg-info text-white",
          },
          "ref": null,
          "rendered": "Make a comment...",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <TextAreaFieldGroup
                  name="text"
                  onChange={[Function]}
                  placeholder="Reply to post"
                  value=""
                />
              </div>
              <button
                className="button-small button-warning"
                type="submit"
              >
                Submit
              </button>
            </form>,
            "className": "card-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="form-group"
                >
                  <TextAreaFieldGroup
                    name="text"
                    onChange={[Function]}
                    placeholder="Reply to post"
                    value=""
                  />
                </div>,
                <button
                  className="button-small button-warning"
                  type="submit"
                >
                  Submit
                </button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <TextAreaFieldGroup
                    name="text"
                    onChange={[Function]}
                    placeholder="Reply to post"
                    value=""
                  />,
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "error": undefined,
                    "name": "text",
                    "onChange": [Function],
                    "placeholder": "Reply to post",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "button-small button-warning",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card card-info"
        >
          <div
            className="card-header bg-info text-white"
          >
            Make a comment...
          </div>
          <div
            className="card-body"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <TextAreaFieldGroup
                  name="text"
                  onChange={[Function]}
                  placeholder="Reply to post"
                  value=""
                />
              </div>
              <button
                className="button-small button-warning"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>,
        "className": "post-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-header bg-info text-white"
            >
              Make a comment...
            </div>,
            <div
              className="card-body"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <TextAreaFieldGroup
                    name="text"
                    onChange={[Function]}
                    placeholder="Reply to post"
                    value=""
                  />
                </div>
                <button
                  className="button-small button-warning"
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>,
          ],
          "className": "card card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Make a comment...",
              "className": "card-header bg-info text-white",
            },
            "ref": null,
            "rendered": "Make a comment...",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <TextAreaFieldGroup
                    name="text"
                    onChange={[Function]}
                    placeholder="Reply to post"
                    value=""
                  />
                </div>
                <button
                  className="button-small button-warning"
                  type="submit"
                >
                  Submit
                </button>
              </form>,
              "className": "card-body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="form-group"
                  >
                    <TextAreaFieldGroup
                      name="text"
                      onChange={[Function]}
                      placeholder="Reply to post"
                      value=""
                    />
                  </div>,
                  <button
                    className="button-small button-warning"
                    type="submit"
                  >
                    Submit
                  </button>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <TextAreaFieldGroup
                      name="text"
                      onChange={[Function]}
                      placeholder="Reply to post"
                      value=""
                    />,
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "error": undefined,
                      "name": "text",
                      "onChange": [Function],
                      "placeholder": "Reply to post",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "className": "button-small button-warning",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
